src/settings/settings.h

```diff
@@ -260,7 +260,11 @@ class SettingsBase : public SettingsBaseVirtual
                 sstream << "\"" << pair.first << "\": {\"default_value\":\"" << out_val << "\"},\n";
             }
         }
-        return sstream.str();
+        std::string out_str = sstream.str();
+        if (!out_str.empty()) {
+            out_str = out_str.substr(0,out_str.size()-2) + "\n";
+        }
+        return out_str;
```

Comment:
I think this is a bit of an ugly way to do it. You could do it by adding `,\n` _before_ the string except for the first iteration.
I don't really mind much, though. This is debugging code.

Please do document what this piece of code is for (or change it).